core/session.rb

```diff
@@ -403,14 +403,23 @@ def up(args)
       $out.warning 'You are using mdbci nodes template. ./mdbci up command doesn\'t supported for this boxes!'
       return 0
     else
-      (1..@attempts.to_i).each { |i|
+      (1..@attempts.to_i).each do |i|
         $out.info 'Bringing up ' + (up_type ? 'node ' : 'configuration ') +
           args + ', attempt: ' + i.to_s
-        $out.info 'Destroying current instance'
-        cmd_destr = 'vagrant destroy --force ' + (up_type ? config[1]:'')
-        exec_cmd_destr = `#{cmd_destr}`
-        $out.info exec_cmd_destr
-        cmd_up = 'vagrant up --destroy-on-error ' + '--provider=' + @nodesProvider + ' ' +
+
+        if i == 1
+          $out.info 'Destroying current instance'
+          cmd_destr = 'vagrant destroy --force ' + (up_type ? config[1]:'')
+          exec_cmd_destr = `#{cmd_destr}`
+          $out.info exec_cmd_destr
+        end
+
+        no_parallel_flag = ""
+        if @nodesProvider == "aws"
+          no_parallel_flag = " --no-parallel "
+        end
+
+        cmd_up = 'vagrant up' + no_parallel_flag + ' --provider=' + @nodesProvider + ' ' +
```

Comment:
All string concatenation should be replaced for #{111} placeholders, and moved to contsants with CAPITAL names. For example:
`VAGRANT_NOT_PARALLEL_COMMAND = 'vagrant up #{no_parallel_flag} --provider=#{@nodesProvider} #{upType} '`